### DISCUSSION TITLE
Gatsby.js scroll position saving equivalent for Next.js

### DISCUSSION THREAD

**Discussion by grikomsn** [vercel/next.js#11741] [Help]
Gatsby.js has a feature where it [saves the scroll position on every visited page](https://www.gatsbyjs.org/docs/browser-apis/#shouldUpdateScroll) which persist for the current session (test by opening any Gatsby page, scroll anywhere, navigate to another page, and go back which will restore the scroll position). This feature has been discussed years ago on #3303 but none are similar what Gatsby does.

Same as the discussion title, is there any equivalent solution for Next.js?

**Comment by erhankaradeniz** [comment]
You could have a look at the SWR package from Zeit, this does include scroll recovery. [https://swr.now.sh/](https://swr.now.sh/)
